autogpt_platform/backend/backend/copilot/bot/adapters/teams/adapter_test.py

```
@@ -0,0 +1,1162 @@
+    "host",
+    [
+        "127.1",  # shorthand: ipaddress rejects it, getaddrinfo resolves it
+        "2130706433",  # integer form of 127.0.0.1
+        "0177.0.0.1",  # octal form of 127.0.0.1
+        "localhost",  # a name, not a literal
+    ],
+)
+async def test_attachment_hosts_that_resolve_inward_are_refused(app_id, host):
+    """The URL text cannot say where a name points — only resolving can."""
+    with patch(f"{_CONFIG_PATH}.allow_unverified_requests", return_value=False):
+        with pytest.raises(ValueError):
```

Ruff (0.16.1)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)